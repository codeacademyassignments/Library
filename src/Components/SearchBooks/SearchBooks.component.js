import React, { Component } from 'react';
import {
  TextInput, Button, ImageBackground,
} from 'react-native';
import PropTypes from 'prop-types';
import style from './SearchBooks.style';
import backgroundImage from '../../Images/library.jpg';

export default class SearchBooks extends Component {
  state = {
    searchKey: '',
  }

  onSubmit = () => {
    const { navigation } = this.props;
    const { searchKey } = this.state;
    navigation.navigate('BooksOnSearchKey', {
      searchKey,
    });
  };

  render() {
    return (
      <ImageBackground source={backgroundImage} style={style.outterMostView}>
        <TextInput onEndEditing={this.onSubmit} blurOnSubmit autoCorrect selectionColor="red" style={style.textInput} placeholder="Enter anything related to book" onChangeText={text => this.setState({ searchKey: text })} />
        <Button color="#841584" style={style.searchButton} title="search" onPress={this.onSubmit} />
      </ImageBackground>
    );
  }
}

SearchBooks.propTypes = {
  navigation: PropTypes.object.isRequired,
};
