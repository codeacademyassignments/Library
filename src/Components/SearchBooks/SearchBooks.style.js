import { StyleSheet } from 'react-native';


export default StyleSheet.create({
  outterMostView: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  textInput: {
    color: 'white',
    fontSize: 20,
    textAlign: 'center',
    width: 300,
    fontWeight: 'bold',
  },
  searchButton: {
    color: 'white',
    backgroundColor: 'white',
  },
});
