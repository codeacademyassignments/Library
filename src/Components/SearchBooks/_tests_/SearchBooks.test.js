import renderer from 'react-test-renderer';
import React from 'react';
import { shallow } from 'enzyme';
import SearchBooks from '../SearchBooks.component';

describe('SearchBooks', () => {
  it('should match with old snap', () => {
    const tree = renderer.create(<SearchBooks navigation={{ navigate: jest.fn() }} />);
    expect(tree).toMatchSnapshot();
  });
});

describe('SearchBooks: TextInput', () => {
  const mockNavigation = jest.fn();
  const searchKey = 'engineering';
  const wrapper = shallow(<SearchBooks navigation={{ navigate: mockNavigation }} />);

  it('should change the state when text input changes', () => {
    expect(wrapper.instance().state.searchKey).toEqual('');
    wrapper.find('TextInput').simulate('changeText', searchKey);
    expect(wrapper.instance().state.searchKey).toEqual(searchKey);
  });

  it('should call mockNavigation on end submitting', () => {
    wrapper.find('TextInput').simulate('endEditing');
    expect(mockNavigation).toHaveBeenCalledWith('BooksOnSearchKey', {
      searchKey,
    });
  });
});

describe('SearchBooks: Button', () => {
  const mockNavigation = jest.fn();
  const searchKey = '';
  const wrapper = shallow(<SearchBooks navigation={{ navigate: mockNavigation }} />);
  it('should call mockNavigation on pressing submit button', () => {
    wrapper.find('Button').simulate('press');
    expect(mockNavigation).toHaveBeenCalledWith('BooksOnSearchKey', {
      searchKey,
    });
  });
});

describe('SearchBooks: onSubmit', () => {
  const mockNavigation = jest.fn();
  const searchKey = 'future';
  const wrapper = shallow(<SearchBooks navigation={{ navigate: mockNavigation }} />);
  it('should call mockNavigation on call to onSubmit() with searchKey', () => {
    wrapper.find('TextInput').simulate('changeText', searchKey);
    wrapper.instance().onSubmit();
    expect(mockNavigation).toHaveBeenCalledWith('BooksOnSearchKey', {
      searchKey,
    });
  });
});
