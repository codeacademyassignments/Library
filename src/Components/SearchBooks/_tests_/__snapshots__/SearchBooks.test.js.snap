// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBooks should match with old snap 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/Images/library.jpg",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <TextInput
    allowFontScaling={true}
    autoCorrect={true}
    blurOnSubmit={true}
    onChangeText={[Function]}
    onEndEditing={[Function]}
    placeholder="Enter anything related to book"
    selectionColor="red"
    style={
      Object {
        "color": "white",
        "fontSize": 20,
        "fontWeight": "bold",
        "textAlign": "center",
        "width": 300,
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#841584",
            },
          ]
        }
      >
        search
      </Text>
    </View>
  </View>
</View>
`;
